sdk/identity/azure-identity/test/ut/managed_identity_credential_test.cpp

```diff
@@ -785,6 +801,50 @@ TEST(ManagedIdentityCredential, CloudShellInvalidUrl)
       {"{\"expires_in\":3600, \"access_token\":\"ACCESSTOKEN1\"}"}));
 }
 
+// This helper creates the necessary directories required for the Azure Arc tests, and returns where
+// we expect the valid arc key to exist.
+std::string CreateDirectoryAndGetValidKeyPath()
+{
+  std::string validKeyPath;
+#if defined(AZ_PLATFORM_LINUX)
+  validKeyPath = "/var/opt/azcmagent/tokens";
+  int result = system(std::string("sudo mkdir -p ").append(validKeyPath).c_str());
+  if (result != 0 && errno != EEXIST)
+  {
+    GTEST_LOG_(ERROR) << "Directory creation failure in an AzureArc test: " << validKeyPath
+                      << " Result: " << result << " Error : " << errno;
+    EXPECT_TRUE(false);
+  }
+  validKeyPath += "/";
+#elif defined(AZ_PLATFORM_WINDOWS)
+  validKeyPath = Azure::Core::_internal::Environment::GetVariable("ProgramData");
+  if (validKeyPath.empty())
+  {
+    GTEST_LOG_(ERROR) << "We can't get ProgramData folder path in an AzureArc test.";
+    EXPECT_TRUE(false);
+  }
+  // Unlike linux, we can't use mkdir on Windows, since it is deprecated. We will use
+  // CreateDirectory instead.
+  // https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/mkdir?view=msvc-170
+  validKeyPath += "\\AzureConnectedMachineAgent";
```

Comment:
Why mix / and \? 

FWIW, Windows deals with both, so the path should probably be consistent.

Reply:
The paths are different between OSes anyway, so I don't think it matters. I decided to keep the "more natural" path on Windows.